conftest.py

```diff
@@ -64,20 +64,33 @@ def pytest_runtest_makereport(item, call):
     setattr(item, "rep_" + rep.when, rep)
 
 
-@pytest.fixture(autouse=True)
-def get_task_logs_on_failure(request):
-    """ Scheduler should be the only task running as root
-    """
-    yield
+def get_rotating_task_log_lines(task_id: str, task_file: str):
+    rotated_filenames = [task_file,]
+    rotated_filenames.extend(['{}.{}'.format(task_file, i) for i in range(1, 10)])
+    for filename in rotated_filenames:
+        lines = get_task_logs_for_id(task_id, filename)
+        if not lines:
+            return
+        yield lines
+
+
+def is_test_failure(request):
     for report in ('rep_setup', 'rep_call', 'rep_teardown'):
         if not hasattr(request.node, report):
             continue
         if not getattr(request.node, report).failed:
             continue
+        return True
+    return False
+
+
+@pytest.fixture(autouse=True)
+def get_task_logs_on_failure(request):
+    yield
+    if is_test_failure(request):
         for task_id in get_task_ids():
             for task_file in ('stderr', 'stdout'):
-                log_name = '{}_{}_{}.log'.format(request.node.name, task_id, task_file)
-                task_logs = get_task_logs_for_id(task_id, task_file=task_file)
-                if task_logs:
+                for log_lines in get_rotating_task_log_lines(task_id, task_file):
+                    log_name = '{}_{}_{}.log'.format(request.node.name, task_id, task_file)
                     with open(log_name, 'w') as f:
```

Comment:
You're right that `'a'` would append. My intent was to open a separate file for each task log. So the names will look like: `<node.name>_<task_id>_stdout.log`, `<node.name>_<task_id>_stdout.1.log`, etc. Good catch!